### PR TITLE
No delay before reconnect

### LINKED ISSUE #51: No delay before reconnect
There is a delay between reconnect attempts, but there shouldn't be any delay before the first reconnect attempt.

This affects eredis and eredis_sub.

### PR DESCRIPTION
**Pull request by zuiderkwast**
When the connection is lost, attempt to reconnect immediately.

Waiting `reconnect_sleep` milliseconds should be only *between*
reconnects attempts, not before the first reconnect attempt.

Fixes #51.

### CODE REVIEW

File: src/eredis_client.erl [hunk 1]
```diff
@@ -559,21 +568,21 @@ maybe_reconnect(Reason,
 %% `{connection_ready, Socket}'.
 reconnect_loop(Client, ReconnectSleep, Host, Port, SocketOptions,
                TlsOptions, ConnectTimeout, AuthCmd, Db) ->
-    receive
-        {'EXIT', Client, Reason} -> exit(Reason)
-    after
-        ReconnectSleep ->
-            Client ! reconnect_attempt,
-            case connect(Host, Port, SocketOptions, TlsOptions, ConnectTimeout,
```
Comment (zuiderkwast) [inline src/eredis_client.erl:567]: Here, I just moved the `receive...after ReconnectSleep` block to just before the recursive call. It was at the beginning of the function. There are no other changes to this function.

**Comment by zuiderkwast** [comment]
@e-kmisk

File: test/eredis_tcp_SUITE.erl [hunk 2]
```diff
@@ -420,10 +420,10 @@ t_unknown_client_cast(Config) when is_list(Config) ->
     ?assertMatch(ok, eredis:stop(C)).
 
 t_tcp_closed(Config) when is_list(Config) ->
-    C = c(),
+    {ok, C} = eredis:start_link([{reconnect_sleep, 10000}]),
     ?assertMatch({ok, _}, eredis:q(C, ["DEL", foo], 5000)),
     tcp_closed_rig(C),
-    timer:sleep(200), %% Wait for reconnection (after ~100ms)
+    timer:sleep(100), % Instant reconnect. No sleep before the first attemt.
```
Comment (e-kmisk) [inline test/eredis_tcp_SUITE.erl:426]: "attempt"

File: src/eredis_client.erl (hunk 1 above)
Comment (e-kmisk) [reply src/eredis_client.erl:567]: Nice solution.

**Review by bjosv — APPROVED** [review]
Looks good

**Pull request merged**